doc/block_roadmap.md

```diff
@@ -0,0 +1,77 @@
+## Overview
```

Comment:
I think we can move this file into `/doc/design/block_execution.md`.

Reply:
Done